components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,61 @@
+"use client";
+
+import { Component } from "react";
+
+import BasePanel from "./Sidebar/panels/BasePanel";
+
+import type { ReactNode, ErrorInfo } from "react";
+
+type Props = {
+  children?: ReactNode;
+  sidebarId:
+    | "friends"
+    | "notifications"
+    | "notificationsAlert"
+    | "playback"
+    | "queue"
+    | "none";
+  title: string;
+};
+
+type State = {
+  hasError: boolean;
+};
+
+class ErrorBoundary extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      hasError: false,
+    };
+  }
+
+  public static getDerivedStateFromError(_: Error): State {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  public componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+    console.error("Uncaught error:", error, errorInfo);
+  }
+
+  public render() {
+    const { hasError } = this.state;
+    const { children } = this.props;
+    const { title } = this.props;
+    const { sidebarId } = this.props;
+    return (
+      <div>
+        {hasError ? (
+          <BasePanel title={title} sidebarId={sidebarId}>
+            <h1>Something went wrong!</h1>
```

Comment:
can you add a try again button that just does `router.refresh()`? wrap the entire class with the `withRouter()` func which passes in `router` as a prop of type `NextRouter`. If you think it's not worth the effort that's ok, just an idea to improve ux